frameworks/PHP/openswoole/openswoole-no-async.dockerfile

```diff
@@ -1,15 +1,21 @@
-FROM php:8.1-cli
+FROM php:8.2-cli
+
+RUN apt-get update && apt-get install -y git > /dev/null
+
+RUN docker-php-ext-install opcache pdo_mysql > /dev/null
 
 RUN pecl install openswoole > /dev/null && \
```

Comment:
I don't really know PHP, but does this pin the version of Openswoole? I noticed that the Dockerfile for the PostgreSQL permutation was using a different approach.

Reply:
Ah, I see, but the main point still stands - does this pin the version of Openswoole?

Reply:
No, pecl don't use semver or an updating system.
Both Swoole and OpenSwoole, release a new version every 2-3 months from 2021.
So they use a 'fail fast' solution.

I try that all the PHP ecosystem in this bench, is updated and working.
But I was busy, the last weeks. Both are failing for the last ~3 runs only.
And a lot of times, the run fail for a fw, is downloading from the servers. Not that the fw is failing, so I wait some fails.

Also with PHP, we use the major version. All the minor versions are automatic.
The same with composer, that we use semantic versioning, and still sometimes fail.

When fail I try to fix it or create an issue in the fw (so the bench also is used to find bugs) and if necessary I marked them as broken.
Without the fail fast way, we'll have very old versions. And use a lot of my free time, now I have near to 500 PRs merged. If a need to update the pecl versions each 2-3 months, I will need 4000 PRs.

No PHP framework stay more than ~3 months failing in the runs.

Only check the work for major PHP versions from 2021: #6184, #6894, #7703 (still updating)

Reply:
You made the PR to mark as broken a lot of frameworks.
I do it normally for the PHP frameworks.

Reply:
Ultimately it is up to the TechEmpower team to set policies, but my impression is that they prefer when all dependency versions are pinned, which also makes it possible to reproduce results. The reason you have opened so many pull requests is because you have changed one or a few implementations in each one, but there is no requirement to do it that way - if you update versions every quarter, you can just do 4 PRs in a year.

Reply:
I was thinking so, the first time that I tried to update a major PHP version.
Some years ago, with few php frameworks, I created 1 PR to update all the PHP frameworks.
It was a nightmare, in almost all aspects.

Reply:
It would be ideal if we could pin the version of OpenSwoole. If you used the same opening lines as the postgres dockerfile at least the compilation step could be cached. Would there be any issues using OpenSwoole with other tests if it's using the configure with postgres option?

Reply:
The mysql variants aren't compiled.
The postgres version is pined, but it don't execute till update the bench config. I send a PR to doubaokun:fix_openswoole branch, but still not merged.

The good think is that now the 2 mysql variants dockerfiles are identical, and so 1 variant use 1s.

Reply:
Now 1s
https://github.com/TechEmpower/FrameworkBenchmarks/actions/runs/4228688851/jobs/7344367870#step:9:1355

Before ~3m 40s each
https://tfb-status.techempower.com/unzip/results.2022-11-13-10-29-24-097.zip/results/20221107183025/openswoole/build/openswoole.log

https://tfb-status.techempower.com/unzip/results.2022-11-13-10-29-24-097.zip/results/20221107183025/openswoole-no-async/build/openswoole-no-async.log